### PR TITLE
Use Comparator factory methods.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Simplify audio and video stream comparisons using the `Comparator` factory methods.
- Use `Comparator.comparingDouble()` in the preference search to avoid boxing.

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/apache/commons/text/similarity/FuzzyScore.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public FuzzyScore(final Locale locale) {
      * @return result score
      * @throws IllegalArgumentException if the term or query is {@code null}
      */
-    public Integer fuzzyScore(final CharSequence term, final CharSequence query) {
+    public int fuzzyScore(final CharSequence term, final CharSequence query) {
```
Comment (Stypox) [inline app/src/main/java/org/apache/commons/text/similarity/FuzzyScore.java:87]: I think this was taken from [here](https://commons.apache.org/sandbox/commons-text/jacoco/org.apache.commons.text.similarity/FuzzyScore.java.html). Does [the apache license](http://www.apache.org/licenses/LICENSE-2.0) allow modifying code? If so, does it require changes to be documented?

**Review by Stypox — CHANGES_REQUESTED** [review]
Code looks good to me

File: app/src/main/java/org/apache/commons/text/similarity/FuzzyScore.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/apache/commons/text/similarity/FuzzyScore.java]: I undid the change.

File: app/src/main/java/org/schabi/newpipe/util/ListHelper.java [hunk 2]
```diff
@@ -374,11 +376,15 @@ static int getHighestQualityAudioIndex(@Nullable final MediaFormat format,
      */
     static int getMostCompactAudioIndex(@Nullable final MediaFormat format,
                                         @Nullable final List<AudioStream> audioStreams) {
-
         return getAudioIndexByHighestRank(format, audioStreams,
                 // The "-" is important -> Compares ascending (first = highest rank)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/ListHelper.java:380]:
```suggestion
                // The "reversed()" is important -> Compares ascending (first = highest rank)
```

**Review by Stypox — CHANGES_REQUESTED** [review]

**Review by Stypox — APPROVED** [review]
I tested the download dialog, also `ListHelper` tests pass, so this is fine

**Pull request merged**